pkg/operator/controller/ingress/status.go

```diff
@@ -674,6 +674,36 @@ func computeIngressProgressingCondition(conditions []operatorv1.OperatorConditio
 				condition.Reason = "ScopeChanged"
 				condition.Message = message
 				condition.Status = operatorv1.ConditionTrue
+			} else {
+				switch platform.Type {
+				case configv1.AWSPlatformType:
+					if _, ok := service.Annotations[AWSLBTypeAnnotation]; ok {
+						if (ic.Status.EndpointPublishingStrategy.LoadBalancer.ProviderParameters != nil && ic.Status.EndpointPublishingStrategy.LoadBalancer.ProviderParameters.AWS != nil && ic.Status.EndpointPublishingStrategy.LoadBalancer.ProviderParameters.AWS.Type != "NLB") && (service.Annotations[AWSLBTypeAnnotation] == "nlb") {
```

Comment:
What about nesting the `if` and doing an `or`? I feel like it might be a little cleaner opposed to having repeated code blocks.
```suggestion
if service.Annotations[AWSLBTypeAnnotation] == AWSNLBAnnotation {
  if ic.Status.EndpointPublishingStrategy.LoadBalancer.ProviderParameters == nil || 
     ic.Status.EndpointPublishingStrategy.LoadBalancer.ProviderParameters.AWS == nil || 
     ic.Status.EndpointPublishingStrategy.LoadBalancer.ProviderParameters.AWS.Type != AWSNetworkLoadBalancer {
 	condition.Reason = "AWSLBTypeChangedToNLBFromClassic"
	condition.Message = message
	condition.Status = operatorv1.ConditionTrue
}
```

Reply:
In the e2e, we updated ingress controller's `endpointPublishingStrategy` spec, then we wait for the annotation to appear on the service. Here we compare the service annotation with the IC status. Which I think it valid, but what about when the IC.spec is updated, but nor the IC status or the service reflect the change yet? Is that a situation that we should update progressing? Or is that transient enough not to care?

Reply:
regarding suggestion for `What about nesting the if and doing an or? I feel like it might be a little cleaner opposed to having repeated code blocks.` it will change the logic. We need to check for the nil value at every level after Provider Params as the go code will crash if nil value is encountered.

Reply:
> In the e2e, we updated ingress controller's `endpointPublishingStrategy` spec, then we wait for the annotation to appear on the service. Here we compare the service annotation with the IC status. Which I think it valid, but what about when the IC.spec is updated, but nor the IC status or the service reflect the change yet? Is that a situation that we should update progressing? Or is that transient enough not to care?

I did try to test that but the annotation in the service gets quickly updated and the progressing status quickly becomes false as the conditions to make progressing true become false very quickly.

Reply:
> regarding suggestion for What about nesting the if and doing an or? I feel like it might be a little cleaner opposed to 
 having repeated code blocks. it will change the logic. We need to check for the nil value at every level after Provider Params as the go code will crash if nil value is encountered.

Agreed, but this is what my suggestion actually does. The code has 3 if conditions with identical outputs. This suggestion is `nil` safe because if the first condition `ic.Status.EndpointPublishingStrategy.LoadBalancer.ProviderParameters == nil` is true, then it won't execute the next `ic.Status.EndpointPublishingStrategy.LoadBalancer.ProviderParameters.AWS == nil`, so on and so forth.

Please double check my work to make sure I'm not missing something, but I'd argue this is more concise and readable. Pseudo Code:
```
if service has NLB annotation {
      if ProviderParameters is nil or ProviderParameters.AWS is nil or ProviderParameters.AWS.type is not NLB {
          set condition to AWSLBTypeChangedToNLBFromClassic
      }
}
```

Reply:
@gcs278 cool I got it :) thanks! made the changes.